pkg/logql/rangemapper.go

```diff
@@ -373,7 +375,15 @@ func (m RangeMapper) mapRangeAggregationExpr(expr *syntax.RangeAggregationExpr,
 		if labelExtractor && vectorAggrPushdown.Operation != syntax.OpTypeSum {
 			return expr
 		}
-		return m.sumOverFullRange(expr, vectorAggrPushdown, syntax.OpRangeTypeCount, rangeInterval, recorder)
+		// rate({app="foo"}[2m]) =>
+		// => (sum without (count_over_time({app="foo"}[1m]) ++ count_over_time({app="foo"}[1m]) offset 1m) / 120)
+		op := syntax.OpRangeTypeCount
+		if expr.Left.Unwrap != nil {
+			// rate({app="foo"} | unwrap bar [2m])
+			// => (sum without (sum_over_time({app="foo"}[1m]) ++ sum_over_time({app="foo"}[1m]) offset 1m) / 120)
+			op = syntax.OpRangeTypeSum
```

Comment:
I'm a bit confused 😕   are we sure this is equivalent to `sum_over_time`? 
In this case, it would also be useful to add a test to `pkg/logql/downstream_test.go` to compare with the normal engine

Reply:
Yes, I should add a downstream test :+1:

Reply:
Instead of counting log lines for the rate, it sums the individual unwrap values.

Reply:
:thinking: Maybe not ...

Reply:
I think I understood unwrapped rate incorrectly. It does not sum the unwrapped values, but treats them as counter, which is treated as reset, when a subsequent value is smaller than the preceding one.
See
https://github.com/grafana/loki/blob/main/pkg/logql/range_vector.go#L227
and
https://github.com/grafana/loki/blob/main/pkg/logql/range_vector.go#L248-L256

Reply:
Waiting for https://github.com/grafana/loki/pull/6361 to revert `rate` calculation.

Reply:
@ssncferreira Rebased this PR with `main` and now the added tests pass :)